tools/src/main/scala/scala/scalanative/io/Classpath.scala

```diff
@@ -0,0 +1,55 @@
+package scala.scalanative.io
+
+import java.net.URI
+import java.nio.ByteBuffer
+import java.nio.file._
+
+import scala.collection.JavaConverters._
+
+/**
+  * A values which represents a member of a classPath.
+  */
+sealed abstract class Classpath[T <: VFile] {
+  def contents: Stream[T]
+  def close(): Unit
+}
+
+/**
+  * A trait for abstracting memory cached file-like object contained in some member of a classpath.
+  */
+trait VFile {
+  def relativePath: Path
+  def bytes: ByteBuffer
+  def uri: URI
+}
+
+final case class LocalClasspath(path: Path) extends Classpath[LazyVFile[LocalClasspath]] {
```

Comment:
If you don't intend to pattern match on this, it shouldn't be a case class. Case classes cause serious difficulties when trying to evolve the API in compatible ways.